### ISSUE TITLE
user specific search parameter

### ISSUE THREAD

**Issue by unmenschlich** [citronalco/calibre-dnb#7]
How looks about user can change search parameter to search with seriesname and seriesnumber or/and Publisher
as example: tit="$seriesname" and tit="$seriesnumber" not (mat=film or mat=music or mat=microfiches or cod=tt)

at the moment dnb plugin dont find all possible with standard search
tit="Private Love Stories 02" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
per="Private Love Stories 02" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)

maybe because dnb online catalog have titles like "Private love stories / Bd. 1. 2009, 1. Aufl. "

because my old issue, i dont know why but https works now for me too

**Comment by citronalco** [comment]
Splitting titles containing a number sounds reasonable:
Searching for `tit="Private Love Stories" AND tit="2" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)`
returns the correct result (leading zeros have to be removed).

Still have to figure out how to implement this without having to rewrite large parts of the code.

**Comment by unmenschlich** [comment]
i use leading zeros like 02 because my ebook reader need it to sort ebooks correctly if series have more as 10 ebooks.
but with own search parameter i can more specific search with publisher or something else.
thanks that you think about my idea

**Comment by citronalco** [comment]
Can you explain a lttle bit more what you mean with user defined search parameters?

**Comment by unmenschlich** [comment]
at the moment dnb plugin search with this 2 parameters
tit="$title" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
per="$title" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)

on dnb plugin settings there can be a few (2-4) textbox lines where your 2 standard search parameter written.
user can change this textbox lines with own search parameter like: tit="$series" AND tit="§seriesnumber" AND vlg="$publisher" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
when plugin search and textbox 1 is not empty than decode it to url format ( " " as %20 ) and use this as search parameters
https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=§textbox1
if nothing found than same with $textbox2
user must only know what variable names to use for title, seriesname, seriesnumber publisher,...

only as idea how it implement this

**Comment by citronalco** [comment]
I spent the last few evenings in front of the computer and cleaned up the plugin's code quite a bit.
Hope I did not break anything. If I did or if the new version gives worse results than the old one for some books let me know!

Besides some other improvements I implemented the title-with-numbers-query mentioned above.
I also thought about the custom search parameters, but decided not to implement it:
The plugin already searches for many variations of the given metadata, and if I understood Calibre's documentation correctly, it's impossible to use any existing metadata besides title, author and identifiers (ISBN, IDN) in a metadata plugin. So no chance to include the publisher into a query.

The new version fairly untested, so it's not yet in Calibre. If you want to try simply download the current "Master" branch as ZIP (https://github.com/citronalco/calibre-dnb/archive/refs/heads/master.zip)
